code/workspaces/web-app/src/containers/stacks/StacksContainer.js

```diff
@@ -64,21 +65,31 @@ class StacksContainer extends Component {
 
   loadStack() {
     // Added .catch to prevent unhandled promise error, when lacking permission to view content
-    return this.props.actions.loadStacksByCategory(this.props.projectKey, this.props.containerType)
+    return this.props.actions.loadStacksByCategory(this.props.projectKey.value, this.props.containerType)
+      .then(() => { this.setUpdateTimeout(); })
+      .catch((() => {}));
+  }
+
+  updateStack() {
+    this.props.actions.updateStacksByCategory(this.props.projectKey.value, this.props.containerType)
       .then(() => {
-        // If project key changes in state, another timer seemed to be being created
-        // rather than replacing the existing one (would still timeout when navigating to
-        // a different page). Clearing the timeout first seems to solve the issue.
-        if (this.timeout) {
-          clearTimeout(this.timeout);
-        }
-        this.timeout = setTimeout(this.loadStack, refreshTimeout);
+        this.setUpdateTimeout();
       })
-      .catch((() => {}));
+      .catch(() => {});
+  }
+
+  setUpdateTimeout() {
+    // If project key changes in state, another timer seemed to be being created
+    // rather than replacing the existing one (would still timeout when navigating to
+    // a different page). Clearing the timeout first seems to solve the issue.
+    if (this.timeout) {
+      clearTimeout(this.timeout);
```

Comment:
Does this work to clear timeouts when we switch to different pages? Should the clear instead be on an unmount?

Reply:
There is also a clear on the unmount (line 97 which isn't shown in the standard diff). That doesn't seem to clear the timeout when changing to another page that renders stacks (I don't know if it doesn't unmount on that change) and that's why this extra clear came into existence. Together they seem to get the job done!